src/spandrel/architectures/CRAFT/__init__.py

```diff
@@ -0,0 +1,83 @@
+import re
+from ..__arch_helpers.state import get_scale_and_output_channels
+from ...__helpers.model_descriptor import (
+    ImageModelDescriptor,
+    SizeRequirements,
+    StateDict
+)
+
+from .arch.CRAFT import CRAFT
+
+def load(state_dict: StateDict) -> ImageModelDescriptor[CRAFT]:
+    # Required values
+    in_chans = state_dict['conv_first.weight'].shape[1]
+    embed_dim = state_dict['layers.0.conv.weight'].shape[1]
+
+    depths = []
+    num_heads = []
+
+    split_size_0 = 4
+    split_size_1 = 16
+
+    mlp_ratio = 2.
+
+    qkv_bias = True
+    qk_scale = None
+
+    # norm_layer=nn.LayerNorm
+
+    upscale, _ = get_scale_and_output_channels(state_dict['upsample.0.bias'].shape[0], in_chans)
```

Comment:
We can actually calculate `upscale` directly here. We know that `state_dict['upsample.0.bias'].shape[0]` is `(scale ** 2) * num_out_ch`, and we know that `num_out_ch == in_chans`.
```suggestion
    upscale = int(math.sqrt(state_dict['upsample.0.bias'].shape[0] / in_chans))
```